Branch-SDK-TestBed/src/io/branch/branchandroiddemo/MainActivity.java

```diff
@@ -309,6 +331,40 @@ public String getSharingMessageForChannel(String channel) {
         //        }catch (JSONException ignore){
         //        }
 
+        // Tracking events
+        findViewById(R.id.cmdTrackCustomEvent).setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                new BranchEvent(BRANCH_STANDARD_EVENT.PURCHASE)
+                        .addCustomProperty("Custom_Event_Property_Key11", "Custom_Event_Property_val11")
+                        .addCustomProperty("Custom_Event_Property_Key22", "Custom_Event_Property_val22")
+                        .logEvent(MainActivity.this);
+            }
+        });
+
+        findViewById(R.id.cmdTrackStandardEvent).setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                new BranchEvent(BRANCH_STANDARD_EVENT.PURCHASE)
+                        .setAffiliation("test_affiliation")
+                        .setCoupon("test_coupon")
+                        .setCurrency(CurrencyType.USD)
+                        .setDescription("Event _description")
+                        .setShipping(10.2)
+                        .setTax(12.3)
+                        .setRevenue(1.5)
+                        .setCondition(BranchEvent.CONDITION.FAIR)
```

Comment:
Whoa. @E-B-Smith are you doing something similar on iOS for this and other constants?